src/java.base/share/classes/sun/security/ec/ECPrivateKeyImpl.java

```diff
@@ -92,48 +92,35 @@ public final class ECPrivateKeyImpl extends PKCS8Key implements ECPrivateKey {
             throws InvalidKeyException {
         this.s = s;
         this.params = params;
-        makeEncoding(s);
-
+        byte[] sArr = s.toByteArray();
+        // convert to fixed-length array
+        int numOctets = (params.getOrder().bitLength() + 7) / 8;
+        byte[] sOctets = new byte[numOctets];
+        int inPos = Math.max(sArr.length - sOctets.length, 0);
+        int outPos = Math.max(sOctets.length - sArr.length, 0);
+        int length = Math.min(sArr.length, sOctets.length);
+        System.arraycopy(sArr, inPos, sOctets, outPos, length);
+        Arrays.fill(sArr, (byte) 0);
+        makeEncoding(sOctets);
+        Arrays.fill(sOctets, (byte) 0);
```

Comment:
You need a try-finally block so that the arrays are cleared even in the case an exception is thrown.